pytripgui/viewcanvas_vc/viewcanvas_cont.py

```diff
@@ -97,25 +96,32 @@ def plot_bg(self):
         chessboard_data = np.add.outer(range(32), range(32)) % 2  # chessboard
         self._ui.plot_bg(chessboard_data)
 
-    def set_patient(self, patient):
+    def set_patient(self, patient, state):
         self._ui.clear()
-        self._model = PlotModel()
+        if state is None:
+            state = ProjectionSelector()
+        self._model = PlotModel(state)
+
         if patient.ctx:
             self._model.set_ctx(patient.ctx)
-
         self._ui.set_position_changed_callback(self.set_current_slice_no)
         self.update_viewcanvas()
 
-    def set_simulation_results(self, simulation_results):
+        return state
```

Comment:
This function should not return object of class "ProjectionSelector" because it is named: "set_patient"
Please add method named "get_gui_state" to this class (ViewCanvasCont) and use it int client code - app_logic